creator-node/src/ipfsClient.js

```diff
@@ -27,69 +25,115 @@ async function logIpfsPeerIds () {
 }
 
 /**
- * Wrapper to ipfs.add() -- This wrapper allows for ipfs calls to be made async or synchronous.
+ * Wrapper to ipfs.add() -- Allows enabling/disabling adding content to the ipfs daemon. Generally used for tracks, metadata.
  * @param {Object} ipfs ipfs library -- can be two different versions (see ipfsClient.js)
- * @param {Buffer|Object[]} inputData either a single buffer, or an Object[] with the structure { path: string, content: buffer }
+ * @param {Buffer} inputData a single buffer input
  * @param {Object?} ipfsConfig ipfs add config options
  * @param {Object?} logContext
- * @param {boolean?} addToIPFSDaemon flag to override adding to ipfs daemon
- * @returns {string|string[]|Object|Object[]} hashes from content addressing fn, or ipfs daemon responses
+ * @param {boolean?} enableIPFSAdd flag to add content to ipfs daemon
+ * @returns {string} hash from content addressing fn, or ipfs daemon response
  */
-async function ipfsAddWrapper (ipfs, inputData, ipfsConfig = {}, logContext = {}, addToIPFSDaemon = false) {
+async function ipfsSingleAddWrapper (ipfs, inputData, ipfsConfig = {}, logContext = {}, enableIPFSAdd = false) {
   const logger = genericLogger.child(logContext)
 
-  // If async ipfs add is enabled, or if `addToIPFSDaemon` flag passed in is false, asynchronously
-  // add to ipfs. Else, synchronously add to ipfs and wait for the response.
-  let ipfsDaemonHashes
-  if (ENABLE_ASYNC_IPFS_ADD && !addToIPFSDaemon) {
-    ipfs.add(inputData, ipfsConfig) // Do not await it
-  } else {
-    ipfsDaemonHashes = await ipfs.add(inputData, ipfsConfig)
+  // Generate hash with ipfs content hashing logic
+  const onlyHash = await ipfsHashOf(inputData)
+
+  // If async ipfs add is enabled, synchronously add to ipfs.
+  let ipfsDaemonHash
+  if (enableIPFSAdd) {
+    try {
+      ipfsDaemonHash = (await ipfs.add(inputData, ipfsConfig))[0].hash
```

Comment:
we might want a timeout here. no guarantee that an operation doesn't hang indefinitely. i'd say like 90 seconds seems reasonable. but that might involve changing the ipfs from ipfsAPI to ipfsLatestAPI

Reply:
just going to do a promise.race with a settimeout